### PR TITLE
Eslint one point oh

### PR DESCRIPTION
**Pull request by anselmbradford**
Updates ESLint to v1.0.0. and fixes errors/warnings.

## Additions

- Adds `restParams: true` to ESLint config, to allow use of ...rest syntax.
- Adds `experimentalObjectRestSpread: false` to ESLint config.
- Adds `qunit: false` to environments.
- Adds `serviceworker: true` to environments to support use of serviceworker API.
- Adds `embertest: false` to environments.
- Adds new rules: `no-implicit-coercion`, `no-invalid-this`, `no-param-reassign`, `no-useless-call`, `callback-return`, `id-length`, `id-match`, `arrow-parens`, `arrow-spacing`, `no-class-assign`, `no-const-assign`, `prefer-spread`, `prefer-reflect`, `require-yield`.

## Removals

- Removed deprecated `no-reserved-keys` rule.

## Changes

- Changes `phantomjs` environment to `false` since this should be overridden if needed in a test directory eslintrc file.
- Fixed linting warnings/errors, some new, some old.
- Added eslint ignore lines where they won't be updated: common.js and jenkins config.

## Testing

- `gulp lint`

## Review

- @KimberlyMunoz 
- @sebworks 

## Todos

- `no-invalid-this` rule is a good best practice, but the existing codebase has far too many invalid uses of this and expect the wider bureau codebase is this way as well. Perhaps after refactoring the JS we can elevate this to a warning level at some point.
- `id-match` rule is potentially very useful for creating consistent naming conventions around CF variables, for example, but the rule would require careful study before enabling.

### CODE REVIEW

File: src/static/js/modules/classes/ContentSlider.js [hunk 1]
```diff
@@ -62,9 +62,9 @@ ContentSlider.prototype.init = function() {
 
 ContentSlider.prototype.slideInContent = function( e ) {
   e.preventDefault();
-  var contents,
-      $div = $( '<div>' ),
-      $node = $( $( e.currentTarget ).data( 'content' ) );
+  var contents;
+  var $div = $( '<div>' );
```
Comment (KimberlyMunoz) [inline src/static/js/modules/classes/ContentSlider.js]: I'm a big fan of this rule change.

File: .eslintrc [hunk 2]
```diff
@@ -238,9 +239,20 @@ rules:
   # Disallows use of leading or trailing decimal points in numeric literals.
   no-floating-decimal: 2
 
+  # Disallow the type conversions with shorter notations.
+  no-implicit-coercion:
+    - 1
+    -
+      boolean: true
```
Comment (sebworks) [inline .eslintrc]: I do love `!!`

**Comment by sebworks** [comment]
:+1:

File: .eslintrc (hunk 2 above)
Comment (anselmbradford) [inline .eslintrc]: I'm okay with updating this to `false` to exclude boolean in the list. The other bit this catches, `var b = ~foo.indexOf(".");`, I've never actually seen in the wild.

@KimberlyMunoz do you have a preference? Warn on `!!var` to update to `Boolean( var )` or not?

Comment (sebworks) [inline .eslintrc]: Boolean is fine @anselmbradford .

Comment (KimberlyMunoz) [inline .eslintrc]: Nah, no preference either way.

**Pull request merged**